rcsbapi/data/schema.py

```diff
@@ -396,6 +415,59 @@ def get_descendant_fields(self, schema_graph, node, visited=None):
+        instances = [
+            "polymer_entity_instances",
+            "branched_entity_instances",
+            "nonpolymer_entity_instances",
+            "polymer_entity_instance",
+            "nonpolymer_entity_instance",
+            "branched_entity_instance"
+        ]
+
+        for single_id in input_ids:
+            if re.match(r"^(MA|AF)_.*_[0-9]+$", single_id) and input_type in entities:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(re.findall(r"^[^_]*_[^_]*", input_ids[0])[0])
+                    inputDict["entity_id"] = str(re.findall(r"^(?:[^_]*_){2}(.*)", input_ids[0])[0])
+            elif (re.match(r"^(MA|AF)_.*\.[A-Z]$", single_id) or re.match(r"^[1-9][A-Z]{3}\.[A-Z]$", single_id)) and input_type in instances:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(re.findall(r"^[^.]+", input_ids[0])[0])
+                    inputDict["asym_id"] = str(re.findall(r"(?<=\.).*", input_ids[0])[0])
+            elif (re.match(r"^(MA|AF)_.*-[0-9]+$", single_id) or re.match(r"^[1-9][A-Z]{3}-[0-9]+$", single_id)) and input_type in ["assemblies", "assembly"]:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(re.findall(r"^[^-]+", input_ids[0])[0])
+                    inputDict["assembly_id"] = str(re.findall(r"[^-]+$", input_ids[0])[0])
+            elif (re.match(r"^(MA|AF)_.*-[0-9]+\.[0-9]+$", single_id) or re.match(r"^[1-9][A-Z]{3}-[0-9]+\.[0-9]+$", single_id)) and input_type in ["interfaces", "interface"]:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(re.findall(r"^[^-]+", input_ids[0])[0])
+                    inputDict["assembly_id"] = str(re.findall(r"-(.*)\.", input_ids[0])[0])
+                    inputDict["interface_id"] = str(re.findall(r"[^.]+$", input_ids[0])[0])
+            elif (re.match(r"^(MA|AF)_.*$", single_id) or re.match(r"^[1-9][A-Z]{3}$", single_id)) and input_type in ["entries", "entry"]:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(input_ids[0])
+            elif re.match(r"^\d{1}[A-Za-z]{3}_\d{1}$", single_id) and input_type in entities:
+                attr_name = [single_id["name"] for single_id in attr_list]
+                if len(input_ids) == 1:
+                    inputDict["entry_id"] = str(re.findall(r"^[^_]+", input_ids[0])[0])
+                    inputDict["entity_id"] = str(re.findall(r"[^_]+$", input_ids[0])[0])
+            else:
+                raise ValueError(f"Invalid ID format: {single_id}")
```

Comment:
This message could be made more informative by included that type that they entered- 
```python
ValueError(f"Invalid ID format for {input_type}: {single_id}")
```